src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2501,21 +2501,15 @@ private function registerNotifierConfiguration(array $config, ContainerBuilder $
             $container->removeDefinition('notifier.channel.email');
         }
 
-        $servicesWithBusArgument = ['texter', 'chatter', 'notifier.channel.chat', 'notifier.channel.email', 'notifier.channel.sms'];
-        if (false === $messageBus = $config['message_bus']) {
-            foreach ($servicesWithBusArgument as $serviceId) {
-                if ($container->hasDefinition($serviceId)) {
-                    $container->getDefinition($serviceId)->replaceArgument(1, null);
-                }
+        foreach (['texter', 'chatter', 'notifier.channel.chat', 'notifier.channel.email', 'notifier.channel.sms'] as $serviceId) {
+            if (!$container->hasDefinition($serviceId)) {
+                continue;
             }
-        } else {
-            foreach ($servicesWithBusArgument as $serviceId) {
-                if ($container->hasDefinition($serviceId)) {
-                    $container->getDefinition($serviceId)
-                        ->setArgument(0, null)
```

Comment:
I don't get why you are removing the transports here. We do need the transports in all cases.

Reply:
Not sure what you mean, but I am not removing the transports. I am adding the message bus service to every channel, in case the `message_bus` config is filled.

But the else branch looks odd. The `message_bus` config value can be a boolean or a service id. I will check: `$container->getDefinition($serviceId)->replaceArgument(1, $messageBus ? new Reference($messageBus) : new Reference('messenger.default_bus', ContainerInterface::NULL_ON_INVALID_REFERENCE));` again.